runtime/src/bank.rs

```diff
@@ -2076,8 +2098,15 @@ impl Bank {
         );
 
         // Add a bogus executable native account, which will be loaded and ignored.
-        let account = native_loader::create_loadable_account(name);
-        self.store_account(&program_id, &account);
+        let account = native_loader::create_loadable_account(
+            name,
+            self.inherit_specially_retained_account_balance(&existing_genuine_program),
```

Comment:
this new `inherit_specially_retained_account_balance` call is justified by this: https://github.com/solana-labs/solana/pull/13884/files#diff-ed47b4a0198313377e091bb3957bbbc63d937805426d1b2b6de39d0a50d32a0cR10122

Reply:
fixed at #14042